### PR TITLE
panel_20

### LINKED ISSUE #270: panel_20
Description: panel plots of different sizes

NCL Script: https://www.ncl.ucar.edu/Applications/Scripts/panel_20.ncl
NCL Plot: https://www.ncl.ucar.edu/Applications/Images/panel_20_lg.png

Category: Panel

### PR DESCRIPTION
**Pull request by hCraker**
Closes #270 
Original NCL plot: http://www.ncl.ucar.edu/Applications/Images/panel_20_lg.png
Current Python plot:
image

### CODE REVIEW

**Comment by hCraker** [comment]
I'd like to choose a better colormap for the contour plots, but I'm not sure what to use. I was thinking of using the diverging `BlueRed` colormap, but I don't know how to force the transition from blue to red to happen at zero. Any ideas of how to do this?

This is what it looks like when I just swap out the colormap
image

**Comment by anissa111** [comment]
> I'd like to choose a better colormap for the contour plots, but I'm not sure what to use. I was thinking of using the diverging `BlueRed` colormap, but I don't know how to force the transition from blue to red to happen at zero. Any ideas of how to do this?

Yeah! You can do this with `DivergingNorm`. See [this matplotlib example].(https://matplotlib.org/3.1.0/gallery/userdemo/colormap_normalizations_diverging.html#sphx-glr-gallery-userdemo-colormap-normalizations-diverging-py)

By adding the line `divnorm = colors.DivergingNorm(vmin=-15, vcenter=0, vmax=40)` under the colormap selection and adding `norm=divnorm` to the contourf calls, I was able to produce the following plot:
image

Is this more like what you had in mind?

File: Plots/Panels/NCL_panel_20.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
+"""
+NCL_panel_20.py
+===============
+This script illustrates the following concepts:
+   - Drawing four different-sized plots on the same page using gridspec
+
+See following URLs to see the reproduced NCL plot & script:
+    - Original NCL script: http://www.ncl.ucar.edu/Applications/Scripts/panel_20.ncl
```
Comment (michaelavs) [inline Plots/Panels/NCL_panel_20.py:8]:
```suggestion
   - Original NCL script: http://www.ncl.ucar.edu/Applications/Scripts/panel_20.ncl
```
Indent is one space too far right for this line and line 9

File: Plots/Panels/NCL_panel_20.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    - Original NCL plot: http://www.ncl.ucar.edu/Applications/Images/panel_20_lg.png
+
+"""
+
+###############################################################################
+# Import packages:
```
Comment (michaelavs) [inline Plots/Panels/NCL_panel_20.py:14]:
```suggestion
# Import packages:

```
Adding line break for Sphinx build

File: Plots/Panels/NCL_panel_20.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+time_1 = ds.isel(time=1).drop_vars('time')
+
+# Ensure longitudes range from 0 to 360 degrees
+U_0 = gvutil.xr_add_cyclic_longitudes(time_0.U, "lon")
+U_1 = gvutil.xr_add_cyclic_longitudes(time_1.U, "lon")
+
+###############################################################################
+# Create helper functions:
+
+
```
Comment (michaelavs) [inline Plots/Panels/NCL_panel_20.py:43]:
```suggestion
```
 Remove extra line break

File: Plots/Panels/NCL_panel_20.py [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+                                     yticks=np.arange(-20, 51, 10),
+                                     xticklabels=['90S', '60S', '30S', '0',
+                                                  '30N', '60N', '90N'])
+
+    # Use geocat.viz.util convenience function to add minor and major ticks
+    gvutil.add_major_minor_ticks(ax,
+                                 x_minor_per_major=3,
+                                 y_minor_per_major=5,
+                                 labelsize=12)
+
```
Comment (michaelavs) [inline Plots/Panels/NCL_panel_20.py:66]:
```suggestion
```
Removing extra line

File: Plots/Panels/NCL_panel_20.py [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    # Remove the degree symbol from tick labels
+    ax.yaxis.set_major_formatter(LatitudeFormatter(degree_symbol=''))
+    ax.xaxis.set_major_formatter(LongitudeFormatter(degree_symbol=''))
+
+    # Use geocat.viz.util convenience function to set titles and labels
+    gvutil.set_titles_and_labels(ax,
+                                 maintitle='300mb',
+                                 maintitlefontsize=8)
+
+
```
Comment (michaelavs) [inline Plots/Panels/NCL_panel_20.py:99]:
```suggestion
```
Removing extra line

File: Plots/Panels/NCL_panel_20.py [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+###############################################################################
+# Plot:
+fig = plt.figure(figsize=(12, 10))
```
Comment (michaelavs) [inline Plots/Panels/NCL_panel_20.py:102]:
```suggestion

fig = plt.figure(figsize=(12, 10))
```
Add extra line break

**Comment by michaelavs** [comment]
I added a couple of formatting fixes, but over all code is running and looking good so far! I did want to ask about if you've done a local build of Sphinx and if it was successful? I ask because I tried to build locally but the code didn't produce anything past the beginning text lines, which I think is just an issue on my side, but wanted to double check.

**Comment by hCraker** [comment]
> Yeah! You can do this with `DivergingNorm`. See [this matplotlib example](https://matplotlib.org/3.1.0/gallery/userdemo/colormap_normalizations_diverging.html#sphx-glr-gallery-userdemo-colormap-normalizations-diverging-py).
> 
> By adding the line `divnorm = colors.DivergingNorm(vmin=-15, vcenter=0, vmax=40)` under the colormap selection and adding `norm=divnorm` to the contourf calls, I was able to produce the following plot:
> image
> 
> Is this more like what you had in mind?

Thank you! This is exactly what I was trying to do.

**Comment by hCraker** [comment]
> I added a couple of formatting fixes, but over all code is running and looking good so far! I did want to ask about if you've done a local build of Sphinx and if it was successful? I ask because I tried to build locally but the code didn't produce anything past the beginning text lines, which I think is just an issue on my side, but wanted to double check.

I have generated the documentation and it looks fine to me. I don't see any glaring issues.

File: Plots/Panels/NCL_panel_20.py [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+
+# Use helper function to create contour plots with identical axes parameters
+format_contour_axes(ax3)
+format_contour_axes(ax4)
+
+# Add left and right titles for both map plots
+ax3.set_title(U_0.long_name, loc='left', y=1.04, fontsize=8)
+ax3.set_title('Time = 0', loc='right', y=1.04, fontsize=8)
+ax4.set_title(U_0.long_name, loc='left', y=1.04, fontsize=8)
+ax4.set_title('Time = 0', loc='right', y=1.04, fontsize=8)
```
Comment (michaelavs) [inline Plots/Panels/NCL_panel_20.py:157]:
```suggestion
ax4.set_title('Time = 1', loc='right', y=1.04, fontsize=8)
```
For consistency with NCL plot

**Comment by michaelavs** [comment]
One last formatting style update and I also now see a warning that DivergingNorm is depreciated after 3.2 and should be replaced with [TwoSlopeNorm](https://matplotlib.org/3.2.1/gallery/userdemo/colormap_normalizations_diverging.html) instead. It looks like it will be a relatively similar implementation process, so a pretty easy update. After those two updates, I am ready to approve this PR!

**Review by michaelavs — APPROVED** [review]
Looks good!

**Review by anissa111 — APPROVED** [review]
This looks good!

**Pull request merged**